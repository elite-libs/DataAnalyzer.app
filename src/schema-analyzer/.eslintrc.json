{
  "parserOptions": {
    "ecmaVersion": 2020,
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "env": {
    "es6": true,
    "node": true,
    "browser": true,
    "jest": true
  },
  "plugins": ["import", "node", "promise", "prettier"],
  "extends": [
    "prettier",
    "prettier/@typescript-eslint",
    "prettier/babel",
    "prettier/react"
  ]
}
