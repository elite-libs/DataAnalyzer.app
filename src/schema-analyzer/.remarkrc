{
  "settings": {
    "listItemIndent": "mixed"
  },
  "plugins": [
    "remark-stringify",
    ["remark-lint-list-item-indent", false]
  ]
}