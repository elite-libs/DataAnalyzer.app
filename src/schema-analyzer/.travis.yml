language: node_js
node_js:
    - 12
install:
    - npm install -g codecov
script:
    - npm ci
    - npm test
    - codecov
