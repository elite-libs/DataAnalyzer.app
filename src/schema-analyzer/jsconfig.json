{
  "compilerOptions": {
    "jsx": "react",
    "checkJs": true,
    "resolveJsonModule": true,
    "esModuleInterop": true
  },
  "typeAcquisition": {
    "include": [
      "jest"
    ]
  }
}