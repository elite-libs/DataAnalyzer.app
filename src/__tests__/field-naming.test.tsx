import eventSearch from '../../public/data/ticketmaster-event-results.json';
import { consolidateNestedTypes, schemaAnalyzer } from '../schema-analyzer/index';

it('can handle _ prefixed names', async () => {
  const results = await schemaAnalyzer('EventResults', eventSearch, {
    strictMatching: false,
    // consolidateTypes: 'field-names',
    debug: true,
  });
  const consolidatedTypes = consolidateNestedTypes(results.flatTypeSummary.nestedTypes!, {
    consolidateTypes: 'field-names',
  });
  // results.nestedTypes = consolidatedTypes.nestedTypes;
  expect(consolidatedTypes.nestedTypes).toMatchSnapshot();
  expect(consolidatedTypes.nestedTypes).toHaveProperty('classifications');
  expect(consolidatedTypes.nestedTypes).toHaveProperty('_links');
});

// it('can transform highly nested types', () => {
//   const typeAliases = {
//     shapeToType: {
//       'next|self': ['EventResults._links'],
//       events: ['EventResults._embedded'],
//       'number|size|totalElements|totalPages': ['EventResults.page'],
//       'href|templated': ['EventResults._links.self', 'EventResults._links.next'],
//       '_embedded|_links|classifications|dates|id|images|locale|name|promoter|sales|test|type|url': [
//         'EventResults._embedded.events',
//       ],
//       'fallback|height|ratio|url|width': [
//         'EventResults._embedded.events.images',
//         'EventResults._embedded.events._embedded.attractions.images',
//       ],
//       public: ['EventResults._embedded.events.sales'],
//       'start|status|timezone': ['EventResults._embedded.events.dates'],
//       'genre|primary|segment|subGenre': [
//         'EventResults._embedded.events.classifications',
//         'EventResults._embedded.events._embedded.attractions.classifications',
//       ],
//       id: [
//         'EventResults._embedded.events.promoter',
//         'EventResults._embedded.events._embedded.venues.markets',
//       ],
//       'attractions|self|venues': ['EventResults._embedded.events._links'],
//       'attractions|venues': ['EventResults._embedded.events._embedded'],
//       'endDateTime|startDateTime|startTBD': [
//         'EventResults._embedded.events.sales.public',
//       ],
//       'dateTBA|dateTBD|localDate|noSpecificTime|timeTBA': [
//         'EventResults._embedded.events.dates.start',
//       ],
//       code: ['EventResults._embedded.events.dates.status'],
//       'id|name': [
//         'EventResults._embedded.events.classifications.segment',
//         'EventResults._embedded.events.classifications.genre',
//         'EventResults._embedded.events.classifications.subGenre',
//         'EventResults._embedded.events._embedded.attractions.classifications.segment',
//         'EventResults._embedded.events._embedded.attractions.classifications.genre',
//         'EventResults._embedded.events._embedded.attractions.classifications.subGenre',
//       ],
//       href: [
//         'EventResults._embedded.events._links.self',
//         'EventResults._embedded.events._links.attractions',
//         'EventResults._embedded.events._links.venues',
//         'EventResults._embedded.events._embedded.venues._links.self',
//         'EventResults._embedded.events._embedded.attractions._links.self',
//       ],
//       '_links|address|city|country|id|locale|location|markets|name|postalCode|state|test|timezone|type': [
//         'EventResults._embedded.events._embedded.venues',
//       ],
//       '_links|classifications|id|images|locale|name|test|type': [
//         'EventResults._embedded.events._embedded.attractions',
//       ],
//       name: ['EventResults._embedded.events._embedded.venues.city'],
//       'name|stateCode': ['EventResults._embedded.events._embedded.venues.state'],
//       'countryCode|name': ['EventResults._embedded.events._embedded.venues.country'],
//       line1: ['EventResults._embedded.events._embedded.venues.address'],
//       'latitude|longitude': ['EventResults._embedded.events._embedded.venues.location'],
//       self: [
//         'EventResults._embedded.events._embedded.venues._links',
//         'EventResults._embedded.events._embedded.attractions._links',
//       ],
//     },
//     typeToShape: {
//       'EventResults._links': 'next|self',
//       'EventResults._embedded': 'events',
//       'EventResults.page': 'number|size|totalElements|totalPages',
//       'EventResults._links.self': 'href|templated',
//       'EventResults._links.next': 'href|templated',
//       'EventResults._embedded.events':
//         '_embedded|_links|classifications|dates|id|images|locale|name|promoter|sales|test|type|url',
//       'EventResults._embedded.events.images': 'fallback|height|ratio|url|width',
//       'EventResults._embedded.events.sales': 'public',
//       'EventResults._embedded.events.dates': 'start|status|timezone',
//       'EventResults._embedded.events.classifications': 'genre|primary|segment|subGenre',
//       'EventResults._embedded.events.promoter': 'id',
//       'EventResults._embedded.events._links': 'attractions|self|venues',
//       'EventResults._embedded.events._embedded': 'attractions|venues',
//       'EventResults._embedded.events.sales.public': 'endDateTime|startDateTime|startTBD',
//       'EventResults._embedded.events.dates.start':
//         'dateTBA|dateTBD|localDate|noSpecificTime|timeTBA',
//       'EventResults._embedded.events.dates.status': 'code',
//       'EventResults._embedded.events.classifications.segment': 'id|name',
//       'EventResults._embedded.events.classifications.genre': 'id|name',
//       'EventResults._embedded.events.classifications.subGenre': 'id|name',
//       'EventResults._embedded.events._links.self': 'href',
//       'EventResults._embedded.events._links.attractions': 'href',
//       'EventResults._embedded.events._links.venues': 'href',
//       'EventResults._embedded.events._embedded.venues':
//         '_links|address|city|country|id|locale|location|markets|name|postalCode|state|test|timezone|type',
//       'EventResults._embedded.events._embedded.attractions':
//         '_links|classifications|id|images|locale|name|test|type',
//       'EventResults._embedded.events._embedded.venues.city': 'name',
//       'EventResults._embedded.events._embedded.venues.state': 'name|stateCode',
//       'EventResults._embedded.events._embedded.venues.country': 'countryCode|name',
//       'EventResults._embedded.events._embedded.venues.address': 'line1',
//       'EventResults._embedded.events._embedded.venues.location': 'latitude|longitude',
//       'EventResults._embedded.events._embedded.venues.markets': 'id',
//       'EventResults._embedded.events._embedded.venues._links': 'self',
//       'EventResults._embedded.events._embedded.venues._links.self': 'href',
//       'EventResults._embedded.events._embedded.attractions.images':
//         'fallback|height|ratio|url|width',
//       'EventResults._embedded.events._embedded.attractions.classifications':
//         'genre|primary|segment|subGenre',
//       'EventResults._embedded.events._embedded.attractions._links': 'self',
//       'EventResults._embedded.events._embedded.attractions.classifications.segment':
//         'id|name',
//       'EventResults._embedded.events._embedded.attractions.classifications.genre':
//         'id|name',
//       'EventResults._embedded.events._embedded.attractions.classifications.subGenre':
//         'id|name',
//       'EventResults._embedded.events._embedded.attractions._links.self': 'href',
//     },
//     shapeAlias: {},
//   };
// });
