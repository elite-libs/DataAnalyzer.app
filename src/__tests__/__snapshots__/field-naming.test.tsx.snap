// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can handle _ prefixed names 1`] = `
Object {
  "EventResults._embedded": Object {
    "fields": Object {
      "events": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 1,
        "nullable": false,
        "type": "Array",
        "typeRef": "EventResults._embedded.events",
        "typeRelationship": "one-to-many",
        "unique": false,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded",
    "totalRows": 1,
  },
  "EventResults._embedded.events": Object {
    "fields": Object {
      "_embedded": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "EventResults._embedded.events._embedded",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "_links": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "EventResults._embedded.events._links",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "classifications": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 1,
        "nullable": false,
        "type": "Array",
        "typeRef": "classifications",
        "typeRelationship": "one-to-many",
        "unique": false,
      },
      "dates": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "EventResults._embedded.events.dates",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "id": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 14,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "images": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 10,
        "nullable": false,
        "type": "Array",
        "typeRef": "images",
        "typeRelationship": "one-to-many",
        "unique": false,
      },
      "locale": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 5,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "name": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 41,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "promoter": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "EventResults._embedded.events.promoter",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "sales": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "EventResults._embedded.events.sales",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "test": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Boolean",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": false,
      },
      "type": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 5,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "url": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 46,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events",
    "totalRows": 1,
  },
  "EventResults._embedded.events._embedded": Object {
    "fields": Object {
      "attractions": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 1,
        "nullable": false,
        "type": "Array",
        "typeRef": "EventResults._embedded.events._embedded.attractions",
        "typeRelationship": "one-to-many",
        "unique": false,
      },
      "venues": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 1,
        "nullable": false,
        "type": "Array",
        "typeRef": "EventResults._embedded.events._embedded.venues",
        "typeRelationship": "one-to-many",
        "unique": false,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events._embedded",
    "totalRows": 1,
  },
  "EventResults._embedded.events._embedded.attractions": Object {
    "fields": Object {
      "_links": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "_links",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "classifications": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 1,
        "nullable": false,
        "type": "Array",
        "typeRef": "classifications",
        "typeRelationship": "one-to-many",
        "unique": false,
      },
      "id": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 11,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "images": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 10,
        "nullable": false,
        "type": "Array",
        "typeRef": "images",
        "typeRelationship": "one-to-many",
        "unique": false,
      },
      "locale": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 5,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "name": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 21,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "test": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Boolean",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": false,
      },
      "type": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 10,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events._embedded.attractions",
    "totalRows": 1,
  },
  "EventResults._embedded.events._embedded.venues": Object {
    "fields": Object {
      "_links": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "_links",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "address": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "EventResults._embedded.events._embedded.venues.address",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "city": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "EventResults._embedded.events._embedded.venues.city",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "country": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "EventResults._embedded.events._embedded.venues.country",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "id": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 12,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "locale": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 5,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "location": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "EventResults._embedded.events._embedded.venues.location",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "markets": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 1,
        "nullable": false,
        "type": "Array",
        "typeRef": "EventResults._embedded.events._embedded.venues.markets",
        "typeRelationship": "one-to-many",
        "unique": false,
      },
      "name": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 20,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "postalCode": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Number",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "state": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "EventResults._embedded.events._embedded.venues.state",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "test": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Boolean",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": false,
      },
      "timezone": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 16,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "type": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 5,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events._embedded.venues",
    "totalRows": 1,
  },
  "EventResults._embedded.events._embedded.venues.address": Object {
    "fields": Object {
      "line1": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 19,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events._embedded.venues.address",
    "totalRows": 1,
  },
  "EventResults._embedded.events._embedded.venues.city": Object {
    "fields": Object {
      "name": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 5,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events._embedded.venues.city",
    "totalRows": 1,
  },
  "EventResults._embedded.events._embedded.venues.country": Object {
    "fields": Object {
      "countryCode": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 2,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "name": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 24,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events._embedded.venues.country",
    "totalRows": 1,
  },
  "EventResults._embedded.events._embedded.venues.location": Object {
    "fields": Object {
      "latitude": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "precision": 10,
        "scale": 8,
        "type": "Float",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "longitude": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "precision": 11,
        "scale": 8,
        "type": "Float",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events._embedded.venues.location",
    "totalRows": 1,
  },
  "EventResults._embedded.events._embedded.venues.markets": Object {
    "fields": Object {
      "id": Object {
        "count": 1,
        "enum": null,
        "identity": true,
        "nullable": false,
        "type": "Number",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events._embedded.venues.markets",
    "totalRows": 1,
  },
  "EventResults._embedded.events._embedded.venues.state": Object {
    "fields": Object {
      "name": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 7,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "stateCode": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 2,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events._embedded.venues.state",
    "totalRows": 1,
  },
  "EventResults._embedded.events._links": Object {
    "fields": Object {
      "attractions": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 1,
        "nullable": false,
        "type": "Array",
        "typeRef": "href",
        "typeRelationship": "one-to-many",
        "unique": false,
      },
      "self": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "href",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "venues": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 1,
        "nullable": false,
        "type": "Array",
        "typeRef": "href",
        "typeRelationship": "one-to-many",
        "unique": false,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events._links",
    "totalRows": 1,
  },
  "EventResults._embedded.events.dates": Object {
    "fields": Object {
      "start": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "EventResults._embedded.events.dates.start",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "status": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "EventResults._embedded.events.dates.status",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "timezone": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 16,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events.dates",
    "totalRows": 1,
  },
  "EventResults._embedded.events.dates.start": Object {
    "fields": Object {
      "dateTBA": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Boolean",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": false,
      },
      "dateTBD": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Boolean",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": false,
      },
      "localDate": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Date",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "noSpecificTime": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Boolean",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": false,
      },
      "timeTBA": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Boolean",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": false,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events.dates.start",
    "totalRows": 1,
  },
  "EventResults._embedded.events.dates.status": Object {
    "fields": Object {
      "code": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 7,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events.dates.status",
    "totalRows": 1,
  },
  "EventResults._embedded.events.promoter": Object {
    "fields": Object {
      "id": Object {
        "count": 1,
        "enum": null,
        "identity": true,
        "nullable": false,
        "type": "Number",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events.promoter",
    "totalRows": 1,
  },
  "EventResults._embedded.events.sales": Object {
    "fields": Object {
      "public": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "EventResults._embedded.events.sales.public",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events.sales",
    "totalRows": 1,
  },
  "EventResults._embedded.events.sales.public": Object {
    "fields": Object {
      "endDateTime": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Date",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "startDateTime": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Date",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "startTBD": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Boolean",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": false,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events.sales.public",
    "totalRows": 1,
  },
  "EventResults._links": Object {
    "fields": Object {
      "next": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "href.templated",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "self": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "href.templated",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._links",
    "totalRows": 1,
  },
  "EventResults.page": Object {
    "fields": Object {
      "number": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Number",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "size": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Number",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "totalElements": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Number",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "totalPages": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Number",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults.page",
    "totalRows": 1,
  },
  "_links": Object {
    "fields": Object {
      "self": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "href",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events._embedded.venues._links",
    "totalRows": 1,
  },
  "classifications": Object {
    "fields": Object {
      "genre": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "id.name",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "primary": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Boolean",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": false,
      },
      "segment": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "id.name",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
      "subGenre": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Object",
        "typeRef": "id.name",
        "typeRelationship": "one-to-one",
        "unique": false,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events.classifications",
    "totalRows": 1,
  },
  "href": Object {
    "fields": Object {
      "href": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 48,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events._links.self",
    "totalRows": 1,
  },
  "href.templated": Object {
    "fields": Object {
      "href": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 44,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "templated": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Boolean",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": false,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._links.self",
    "totalRows": 1,
  },
  "id.name": Object {
    "fields": Object {
      "id": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 18,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "name": Object {
        "count": 1,
        "enum": null,
        "identity": false,
        "length": 6,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events.classifications.segment",
    "totalRows": 1,
  },
  "images": Object {
    "fields": Object {
      "fallback": Object {
        "count": 10,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Boolean",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": false,
      },
      "height": Object {
        "count": 10,
        "enum": null,
        "identity": false,
        "nullable": false,
        "type": "Number",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "ratio": Object {
        "count": 10,
        "enum": null,
        "identity": false,
        "length": 4,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": false,
      },
      "url": Object {
        "count": 10,
        "enum": null,
        "identity": false,
        "length": 106,
        "nullable": false,
        "type": "String",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": true,
      },
      "width": Object {
        "count": 10,
        "enum": null,
        "identity": true,
        "nullable": false,
        "type": "Number",
        "typeRef": undefined,
        "typeRelationship": undefined,
        "unique": false,
      },
    },
    "nestedTypes": Object {},
    "schemaName": "EventResults._embedded.events.images",
    "totalRows": 10,
  },
}
`;
